tools/sgnpmartifactregistryauth/tools.go

```diff
@@ -0,0 +1,26 @@
+package sgnpmartifactregistryauth
+
+import (
+	"context"
+	"os/exec"
+
+	"go.einride.tech/sage/sg"
+)
+
+const (
+	name = "npm-artifact-registry-auth"
+)
+
+func Command(ctx context.Context, args ...string) *exec.Cmd {
+	sg.Deps(ctx, PrepareCommand)
+	return sg.Command(ctx, sg.FromBinDir(name), args...)
+}
+
+func PrepareCommand(ctx context.Context) error {
+	sg.Logger(ctx).Println("authenticating npm to artifact registry...")
+	return sg.Command(
+		ctx,
+		"npx",
+		"google-artifactregistry-auth@3.1.2",
```

Comment:
Maybe break the version into a constant and do something like

```suggestion
		fmt.Sprintf("google-artifactregistry-auth@%s", version),
```

Reply:
Sounds good!